lib/xcflushd/priority_auth_renewer.rb

```diff
@@ -173,53 +173,33 @@ def channel_for_combination(combination)
                                             combination[:metric])
     end
 
-    def publish_auth_repeatedly(combination, authorization)
-      # There is a race condition here. A renew and publish task is only run
-      # when there is not another one renewing the same combination. When there
-      # is another, the incoming request does not trigger a new task, but waits
-      # for the publish below. The request could miss the published message
-      # if events happened in this order:
-      #   1) The request publishes the combination it needs in the requests
-      #      channel.
-      #   2) A new task is not executed, because there is another renewing
-      #      the same combination.
-      #   3) That task publishes the result.
-      #   4) The request subscribes to receive the result, but now it is
-      #      too late.
-      # I cannot think of an easy way to solve this. There is some time
-      # between the moment the requests performs the publish and the
-      # subscribe actions. To mitigate the problem we can publish several
-      # times during some ms. We will see if this is good enough.
-      # Trade-off: Waiting long times between publishing attempts reduces the
-      # probability of triggering the problem described. However, it also makes
-      # incoming requests slow because tasks accumulate in the thread pool
-      # waiting.
+    def publish_auth(combination, authorization)
+      msg = if authorization.authorized?
```

Comment:
I see that you changed this so it's calculated only once :+1: 
How about extracting this to its own method? something like `msg_to_publish(authorization)` ?